polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,293 @@
+        append=True,
+        compute=True,
+        **kwargs,
+    ):
+        """Save hdf5 datasets."""
+
+        compression = kwargs.pop("compression", None) if "compression" in kwargs else None
+        if compression == "none":
+            compression = None
+
+        add_geolocation = kwargs.pop("add_geolocation") if "add_geolocation" in kwargs else False
+
+        # will this be written to one or multiple files?
+        output_names = []
+
+        for dataset_id in dataset:
+            args = self._output_file_kwargs(dataset_id, dtype)
+            out_filename = filename or self.get_filename(**args)
+            output_names.append(out_filename)
+        one_file = all_equal(output_names)
+
+        filename = output_names[0]
+        if not all_equal(output_names):
+            LOG.warning("Writing to {}".format(filename))
+
+        hdf5_fh = self.open_hdf5_filehandle(filename, append=append)
+
+        datasets_by_area = self.iter_by_area(dataset)
+        for area, data_arrs in datasets_by_area:
+            # Initialize source/targets at start of each new AREA grouping.
+            dsets = []
+            targets = []
+            # open hdf5 file handle, check if group already exists.
+            parent_group = self.create_proj_group(filename, hdf5_fh, area)
+
+            if add_geolocation:
+                chunks = data_arrs[0].chunks
+                geo_sets, fnames = self.write_geolocation(
+                    hdf5_fh, filename, parent_group, area, dtype, append, compression, chunks
+                )
+                dsets.append(geo_sets)
+                targets.append(fnames)
+
+            for data_arr in data_arrs:
+                d_dtype = data_arr.dtype if dtype is None else dtype
+                hdf_subgroup = "{}/{}".format(parent_group, data_arr.attrs["name"])
+
+                file_var = FakeHDF5(filename, hdf_subgroup)
+                arr_dset = self.check_variable(hdf5_fh, hdf_subgroup, data_arr.data, d_dtype, append, compression)
+                dsets.append(data_arr.data)
+                targets.append(file_var)
+        return (dsets, targets)
+
+
+def add_writer_argument_groups(parser, group=None):
+    """Create writer argument groups."""
+    from argparse import SUPPRESS
+
+    if group is None:
+        group = parser.add_argument_group(title="hdf5 Writer")
+    group.add_argument("--output-filename", dest="filename", help="Custom file pattern to save dataset to")
+    group.add_argument(
+        "--dtype",
+        choices=NumpyDtypeList(NUMPY_DTYPE_STRS),
+        type=str_to_dtype,
+        help="Data type of the output file (8-bit unsigned " "integer by default - uint8)",
+    )
+    group.add_argument(
+        "--fill-value", dest="fill_value", type=int_or_float, help="Fill value for invalid data in this dataset."
+    )
+    group.add_argument("--compress", default="LZW", help="File compression algorithm (DEFLATE, LZW, NONE, etc)")
+    group.add_argument(
+        "--chunks",
```

Comment:
I don't think you ever use this, right?

Reply:
Correct, however, will this be a command line argument for the future?  Should I actually try to use a user input chunk size, or is P2G always going to use a default chunk size?

Reply:
I think this is referring to the HDF5 chunk size (internal to the HDF5 file). Especially if this came from the old writer then this has nothing to do with dask (we didn't use dask then).

Reply:
My question is directed more towards using dask chunk sizes.  Correct me if I misunderstand, if I send the chunk size to da.store, won't dask control the chunk size written to the HDF5 file?  If that is the case, did you want to see an option in the future for controlling that chunk size, or just use the default chunk size?

Reply:
There will likely not be a command line argument to control dask array chunk size. If anything, it will be an environment variable (inline with Satpy's 'Config' object and its usage). If you'd like to rename this to `--hdf5-chunks` to avoid confusion I am ok with that.

Oh! I just reread the old writer code. You added this `--chunks` didn't you? In that case, please remove this as an option. I thought this was controlling the chunk size of the HDF5 variables on disk.